src/cargo/util/errors.rs

```diff
@@ -300,6 +302,140 @@ impl From<clap::Error> for CliError {
+            //
+            // If you are updating them, please keep the commit ref in the doc comment
+            // of this method up-to-date.
+            r#"!.*\[remote rejected\].*\(error in hook\)"#,
+            r#"!.*\[remote rejected\].*\(failed to lock\)"#,
+            r#"!.*\[remote rejected\].*\(error in Gerrit backend\)"#,
+            r#"remote error: Internal Server Error"#,
+            r#"fatal: Couldn't find remote ref "#,
+            r#"git fetch_pack: expected ACK/NAK, got"#,
+            r#"protocol error: bad pack header"#,
+            r#"The remote end hung up unexpectedly"#,
+            r#"The remote end hung up upon initial contact"#,
+            r#"TLS packet with unexpected length was received"#,
+            r#"RPC failed; result=\d+, HTTP code = \d+"#,
+            r#"Connection timed out"#,
+            r#"repository cannot accept new pushes; contact support"#,
+            r#"Service Temporarily Unavailable"#,
+            r#"The service is currently unavailable"#,
+            r#"Connection refused"#,
+            r#"The requested URL returned error: 5\d+"#,
+            r#"Operation too slow"#,
+            r#"Connection reset by peer"#,
+            r#"Unable to look up"#,
+            r#"Couldn't resolve host"#,
+            r#"Unknown SSL protocol error"#,
+            r#"Revision .* of patch set \d+ does not match refs/changes"#,
+            r#"Git repository not found"#,
+            r#"Couldn't connect to server"#,
+            r#"transfer closed with outstanding read data remaining"#,
+            r#"Access denied to"#,
+            r#"The requested URL returned error: 429"#,
+            r#"RESOURCE_EXHAUSTED"#,
+            r#"Resource has been exhausted"#,
+            r#"check quota"#,
+            r#"fetch-pack: protocol error: bad band #\d+"#,
+            r#"The requested URL returned error: 400"#,
+            r#"fetch-pack: fetch failed"#,
+            r#"fetch-pack: unable to spawn http-fetch"#,
+            r#"fetch-pack: expected keep then TAB at start of http-fetch output"#,
+            r#"fetch-pack: expected hash then LF at end of http-fetch output"#,
+            r#"fetch-pack: unable to finish http-fetch"#,
+            r#"fetch-pack: pack downloaded from .* does not match expected hash .*"#,
+            // Regexes that are not included in chromium source code
+            // This one was reported in https://github.com/rust-lang/cargo/issues/9820
+            r#"Failed to connect .* Timed out"#,
+        ];
+
+        /// Merge all of the regex into a single regex OR-ed together
+        fn merge_regex(regexps: &[&str]) -> regex::Regex {
+            let source = regexps
+                .iter()
+                .format_with("|", |regex, f| f(&format_args!("(?:{})", regex)));
+
+            let source = &format!("(?i){}", source);
+
+            regex::Regex::new(source).expect("BUG: invalid git transient error regex")
```

Comment:
I think this may be a use-case for [`RegexSet`](https://docs.rs/regex/1.5.4/regex/struct.RegexSet.html)?